metadata-ingestion/src/datahub/ingestion/source/powerbi.py

```diff
@@ -1029,6 +1100,16 @@ def json_to_dataset_map(scan_result: dict) -> dict:
                         f"Processing tables attribute for dataset {dataset_instance.name}({dataset_instance.id})"
                     )
 
+                    if not dataset_dict["tables"]:
```

Comment:
So this only happens in ODBC cases? Otherwise tables is always populated?

Reply:
Currently what we've experienced, it has been that only ODBC sources are missing tables. When reflecting this against the [this PR](https://github.com/datahub-project/datahub/pull/6549) it might be that the tables are not there, but the expression is. IMO correct approach would be to use the DAX query as last resort if no other mean is available for discovering the schema.

The API does throw an error 500 sometimes, so it has not been completely reliable API.

Would it be better if this feature would be completely behind a feature flag?